kpatch-build/kpatch-build

```diff
@@ -816,9 +855,30 @@ fi
 echo "Building original source"
 [[ -n "$OOT_MODULE" ]] || ./scripts/setlocalversion --save-scmversion || die
 unset KPATCH_GCC_TEMPDIR
+
+declare -a MAKEVARS
+if [ "$CONFIG_CC_IS_CLANG" -eq 1 ]; then
+	# CROSS_COMPILE does a bit more than just adding a prefix when used
+	# with clang, so we have to do this manually.
+
+	MAKEVARS+=("CC=$TOOLSDIR/kpatch-cc clang")
+
+	if [ -n "$LLVM" ]; then
+		MAKEVARS+=("LD=$TOOLSDIR/kpatch-cc ld.lld")
+	elif [ -n "$LD" ]; then
+		MAKEVARS+=("LD=$TOOLSDIR/kpatch-cc $LD")
+	else
+		# Defaulting to ld, still need to add kpatch-cc prefix
+		MAKEVARS+=("LD=$TOOLSDIR/kpatch-cc ld")
```

Comment:
> Instead of all this, I think you can just check CONFIG_LD_IS_LLD.

Oh, didn't realize it was available through config as well. Thanks.

> Should also set HOSTCC=clang? I had to add that when building in a container without gcc.

Hm, yes. HOSTCC and HOSTLD as well.